python/deltalake/table.py

```diff
@@ -183,13 +183,15 @@ def pyarrow_schema(self) -> pyarrow.Schema:
         """
         return pyarrow_schema_from_json(self._table.arrow_schema_json())
 
-    def to_pyarrow_dataset(self) -> pyarrow.dataset.Dataset:
+    def to_pyarrow_dataset(self, partitions: Optional[List[Tuple]] = []) -> pyarrow.dataset.Dataset:
```

Comment:
Although we are not mutating the partitions here, it's best to avoid the anti-pattern of using mutable object as the default in python, see https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments.

Reply:
I first hit that with Python 2.3, It's been over a decade and they still haven't fixed that bug? :facepalm:

Reply:
this is part of the language design unfortunately, so it probably will never get "fixed" :) at least it won't change in python 3 because it will break existing code.